AEPLifecycle/Tests/XDMLanguageTests.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ Copyright 2022 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+@testable import AEPLifecycle
+import XCTest
+
+class XDMLanguageTests: XCTestCase {
+
+    override func setUp() {
+        continueAfterFailure = true
+    }
+
+    // MARK: Encodable tests
+
+    func testEncodeLanguage_whenValid() throws {
+        // setup
+        let lang = XDMLanguage(language: "en-US")
+        XCTAssertEqual("en-US", lang.language)
+
+        // test
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let data = try XCTUnwrap(encoder.encode(lang))
+        let dataStr = try XCTUnwrap(String(data: data, encoding: .utf8))
+
+        // verify
+        let expected = """
+        {
+          "language" : "en-US"
+        }
+        """
+
+        XCTAssertEqual(expected, dataStr)
+    }
+
+    func testEncodeLanguage_whenInvalid() throws {
+        // setup
+        let lang = XDMLanguage(language: "en-US@calendar=buddhist")
+        XCTAssertNil(lang.language)
+
+        // test
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let data = try XCTUnwrap(encoder.encode(lang))
+        let dataStr = try XCTUnwrap(String(data: data, encoding: .utf8))
+
+        // verify
+        let expected = """
+        {
+
+        }
+        """
+
+        XCTAssertEqual(expected, dataStr)
+    }
+
+    // MARK: isValidLanguageTag tests
+
+    // List of test langauge strings to verify isValidLanguageTag function
+    // [$0: language string to test, $1: expected is valid]
```

Comment:
you can increase readability by splitting this into two lists validValues / invalidValues and having two tests for them